middle_end/flambda2/compare/compare.ml

```diff
@@ -967,9 +967,9 @@ let apply_exprs env apply1 apply2 : Expr.t Comparison.t =
          (Apply.inlining_state apply1)
          (Apply.inlining_state apply2)
     && Apply.Position.equal (Apply.position apply1) (Apply.position apply2)
-    && Flambda_arity.With_subkinds.equal (Apply.args_arity apply1)
+    && Flambda_arity.equal_ignoring_subkinds (Apply.args_arity apply1)
          (Apply.args_arity apply2)
-    && Flambda_arity.With_subkinds.equal
+    && Flambda_arity.equal_ignoring_subkinds
```

Comment:
Curious why we're ignoring subkinds here? That's a change from the earlier semantics, no?